src/MongoDB.Driver/MongoDatabaseImpl.cs

```diff
@@ -203,7 +201,8 @@ public override IAsyncCursor<BsonDocument> ListCollections(IClientSessionHandle
             Ensure.IsNotNull(session, nameof(session));
             options = options ?? new ListCollectionsOptions();
             var operation = CreateListCollectionsOperation(options);
-            return ExecuteReadOperation(session, operation, ReadPreference.Primary, cancellationToken);
+            var effectiveReadPreference = ReadPreferenceResolver.GetEffectiveReadPreference(session, null, ReadPreference.Primary);
```

Comment:
Note that the explicitReadPreference is null, and the defaultReadPreference is Primary (not the ReadPreference in the MongoDatabaseSettings).